pulsar-broker/src/main/java/org/apache/pulsar/client/impl/RawBatchMessageContainerImpl.java

```diff
@@ -187,6 +187,7 @@ public ByteBuf toByteBuf() {
         idData.writeTo(buf);
         buf.writeInt(metadataAndPayload.readableBytes());
         buf.writeBytes(metadataAndPayload);
+        metadataAndPayload.release();
```

Comment:
@heesung-sn how did you find this issue? I think this could deserve a separate PR for maintenance branches? Please report a separate issue in apache/pulsar issues to track this.

Reply:
Yes, actually, I am trying to see if I can cherry-pick this PR to the branch-3.0 and others.

If not easy, I will raise separate PRs to them.